stellargraph/layer/graphsage.py

```diff
@@ -119,9 +119,300 @@ def call(self, x, **kwargs):
+            shape=(self.hidden_dim, self.half_output_dim),
+            initializer=self._initializer,
+            trainable=True,
+        )
+        self.w_self = self.add_weight(
+            name="w_self",
+            shape=(input_shape[0][2], self.half_output_dim),
+            initializer=self._initializer,
+            trainable=True,
+        )
+        if self.has_bias:
+            self.bias = self.add_weight(
+                name="bias",
+                shape=(self.output_dim,),
+                initializer="zeros",
+                trainable=True,
+            )
+        super().build(input_shape)
+
+    def call(self, x, **kwargs):
+        """
+        Apply max aggregation on input tensors, x
+
+        Args:
+          x: Keras Tensor
+
+        Returns:
+            Keras Tensor representing the aggregated embeddings in the input.
+
+        """
+        # x[0]: self vector (batch_size, head size, feature_size)
+        # x[1]: neighbour vector (batch_size, head size, neighobours, feature_size)
+        xw_neigh = K.dot(x[1], self.w_pool) + self.b_pool
+
+        # Take max over neighbour
+        neigh_agg = K.max(self.hidden_act(xw_neigh), axis=2)
+
+        # Apply separate self & neighbour weights and concatenate
+        from_self = K.dot(x[0], self.w_self)
+        from_neigh = K.dot(neigh_agg, self.w_neigh)
+        h_out = K.concatenate([from_self, from_neigh], axis=2)
+
+        if self.has_bias:
+            h_out = self.act(h_out + self.bias)
+        else:
+            h_out = self.act(h_out)
+
+        return h_out
+
+    def compute_output_shape(self, input_shape):
+        """
+        Computes the output shape of the layer.
+        Assumes that the layer will be built to match that input shape provided.
+
+        Args:
+            input_shape (tuple of ints)
+                Shape tuples can include None for free dimensions, instead of an integer.
+
+        Returns:
+            An input shape tuple.
+        """
+        return input_shape[0][0], input_shape[0][1], self.output_dim
+
+
+class MeanPoolAggregator(Layer):
+    """
+    Mean Pool Aggregator for GraphSAGE implemented with Keras base layer
+
+    Args:
+        output_dim (int): Output dimension
+        bias (bool): Optional bias
+        act (Callable or str): name of the activation function to use (must be a
+            Keras activation function), or alternatively, a TensorFlow operation.
+
+    """
+
+    def __init__(
+        self,
+        output_dim: int = 0,
+        bias: bool = False,
+        act: Callable or AnyStr = "relu",
+        **kwargs
+    ):
+        self.output_dim = output_dim
+        assert output_dim % 2 == 0
+        self.half_output_dim = int(output_dim / 2)
+
+        # TODO: These should be user parameters
+        self.hidden_dim = output_dim
+        self.hidden_act = activations.get("relu")
+
+        self.has_bias = bias
+        self.act = activations.get(act)
+        self.w_neigh = None
```

Comment:
self.w_neigh doesn't seem to be used

Reply:
True! I renamed what I used in `build(...)` and forgot to update the code in `__init__`.

Reply:
So now that MeanPoolAggregator is changed to match the structure of MaxPoolAggregator, this no longer applies, right?

Reply:
Yes – this no longer applies.